aasemble/django/apps/api/v1/views.py

```diff
@@ -105,9 +105,15 @@ class SnapshotViewSet(viewsets.ModelViewSet):
             serializer_class = selff.serializers.SnapshotSerializer
 
             def get_queryset(self):
-                if self.request.user.is_superuser:
-                    return self.queryset.all()
-                return self.queryset.filter(mirrorset__owner_id=self.request.user.id)
+                tag=self.request.query_params.get('tag', None)
+                if tag is not None:
+                    if self.request.user.is_superuser:
+                        return self.queryset.all(tags__tag=tag)
+                    return self.queryset.filter(mirrorset__owner_id=self.request.user.id, tags__tag=tag)
+                else:
+                    if self.request.user.is_superuser:
+                        return self.queryset.all()
+                    return self.queryset.filter(mirrorset__owner_id=self.request.user.id)
```

Comment:
I would flip this whole thing upside down. Start by constructing a queryset based on whether self.request.user is superuser or not:

    if self.request.user.is_superuser:
        qs = self.queryset.all()
    else:
        qs = self.queryset.filter(<filter by user id here>)

    tag = self.request.query_params.get('tag', None)
    if tag is not None:
        qs.filter(<whatever you need to do to filter by the tag>)

    return qs